quarkus/tests/integration/src/main/java/org/keycloak/it/utils/KeycloakDistribution.java

```diff
@@ -0,0 +1,337 @@
+                }
+
+                connection.setReadTimeout((int) getStartTimeout());
+                connection.setConnectTimeout((int) getStartTimeout());
+                connection.connect();
+
+                if (connection.getResponseCode() == 200) {
+                    LOGGER.infof("Keycloak is ready at %s", contextRoot);
+                    break;
+                }
+            } catch (Exception ignore) {
+            } finally {
+                if (connection != null) {
+                    connection.disconnect();
+                }
+            }
+        }
+    }
+
+    private long getStartTimeout() {
+        return TimeUnit.SECONDS.toMillis(120);
+    }
+
+    private HostnameVerifier createInsecureHostnameVerifier() {
+        return new HostnameVerifier() {
+            @Override
+            public boolean verify(String s, SSLSession sslSession) {
+                return true;
+            }
+        };
+    }
+
+    private SSLSocketFactory createInsecureSslSocketFactory() throws IOException {
+        TrustManager[] trustAllCerts = new TrustManager[] {new X509TrustManager() {
+            public void checkClientTrusted(final X509Certificate[] chain, final String authType) {
+            }
+
+            public void checkServerTrusted(final X509Certificate[] chain, final String authType) {
+            }
+
+            public X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+        }};
+
+        SSLContext sslContext;
+        SSLSocketFactory socketFactory;
+
+        try {
+            sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null, trustAllCerts, new SecureRandom());
+            socketFactory = sslContext.getSocketFactory();
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+            throw new IOException("Can't create unsecure trust manager");
+        }
+        return socketFactory;
+    }
+
+    private boolean isRunning() {
+        return keycloak != null && keycloak.isAlive();
+    }
+
+    private void asyncReadOutput() {
+        shutdownOutputExecutor();
+        outputExecutor = Executors.newSingleThreadExecutor();
+        outputExecutor.execute(this::readOutput);
+    }
+
+    private void shutdownOutputExecutor() {
+        if (outputExecutor != null) {
+            outputExecutor.shutdown();
+            try {
+                outputExecutor.awaitTermination(30, TimeUnit.SECONDS);
+            } catch (InterruptedException cause) {
+                throw new RuntimeException("Failed to terminate output executor", cause);
+            } finally {
+                outputExecutor = null;
+            }
+        }
+    }
+
+    private void reset() {
+        outputStream.clear();
+        errorStream.clear();
+        exitCode = -1;
+        keycloak = null;
+        shutdownOutputExecutor();
+    }
+
+    private Path prepareDistribution() {
```

Comment:
again, we are repeating this logic in many different places, using the Docker image would simplify